include/rpm/rpmio.h

```diff
@@ -60,11 +60,63 @@ off_t Ftell(FD_t fd);
 int Fclose( FD_t fd);
 
 /** \ingroup rpmio
+ * fdopen(3) clone.
+ *
+ * See Fopen() for details.
  */
 FD_t	Fdopen(FD_t ofd, const char * fmode);
 
 /** \ingroup rpmio
  * fopen(3) clone.
+ *
+ * Supports compression.
+ *
+ * The `fmode` parameter is based on that of `fopen(3)`, but may also include a
+ * compression method (`type` and `flags`) to use when opening the stream, and
+ * has the following format:
+ *
+ * ```
+ * <mode>[flags].<type>
+ * ```
+ *
+ * The compression `type` (compressor) is mandatory, determines the supported
+ * `mode` chars (also mandatory), and can be one of the following:
+ * 
+ * | Type	| Description	    | Mode chars    |
+ * |------------|-------------------|---------------|
+ * | `ufdio`	| no compression    | `r,w,a,b,+,x` |
+ * | `gzdio`	| gzip		    | `r,w,a`	    |
+ * | `bzdio`	| bzip2             | `r,w,a`	    |
+ * | `xzdio`	| xz                | `r,w,a`	    |
+ * | `lzdio`	| lzma (legacy)     | `r,w,a`	    |
+ * | `zstdio`	| zstd              | `r,w,a`	    |
+ *
+ * Compression `flags` must be listed in the following order and can be any of:
+ * 
+ * | Flag	| Description				    | Types		    |
+ * |------------|-------------------------------------------|-----------------------|
+ * | `0-9`	| compression level			    | all except `ufdio`    |
+ * | `T<0-N>`	| no. of threads (0 = autodetect)	    | `xzdio` and `zstdio`  |
```

Comment:
Another reason is that we only support the no-argument variant with *some* compressors so it would be cumbersome to document that, too.

Reply:
Another reason is that we only support the no-argument variant with *some* compressors (xz) so it would be cumbersome to document that difference, too.

Reply:
... maybe, if we *really* wanted the no-argument variant, we could fix the other compressor (zstd), but at that point, I think we're overreacting here :smile: